src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableInterlocked.cs

```diff
@@ -12,6 +12,48 @@ namespace System.Collections.Immutable
     /// </summary>
     public static class ImmutableInterlocked
     {
+        /// <summary>
```

Comment:
This came up in #318 as well. I think we agreed that `Interlocked` itself might be a better place, but that's not available yet. Also, in defense of doing it now with what we have, I've used this API a lot (as we have a copy in VS) and I _always_ use it in connection with immutable collections and nothing else. It's rare to have a transform function that doesn't have side effects in other contexts, I guess.